modules/components/src/NestedTreeView/NestedTreeView.js

```diff
@@ -2,26 +2,44 @@ import React from 'react';
 import ReactTreeView from './ReactTreeView';
 import './style.css';
 
-const NestedTreeView = ({ dataSource, depth = 0 }) =>
-  dataSource.map(
-    ({ title, id, children, isHeader }, i) =>
-      children ? (
-        <ReactTreeView
-          key={id || `${title}-${i}`}
-          nodeLabel={title}
-          defaultCollapsed={true}
-          itemClassName={depth == 0 && 'header'}
-        >
-          <NestedTreeView dataSource={children} depth={depth + 1} />
-        </ReactTreeView>
-      ) : (
-        <div
-          key={id || `${title}-${i}`}
-          className={`tree-view_children leaf ${depth == 0 && 'header'}`}
-        >
-          {title}
-        </div>
-      ),
-  );
+const NestedTreeView = ({
+  dataSource,
+  depth = 0,
+  onLeafSelect = () => {},
+  selectedPath = '',
+}) =>
+  dataSource.map(({ title, id, children, path }, i) => {
+    const selectedPathArray = selectedPath?.split('.') || [];
```

Comment:
It won't make a different right now, but  this will result in `['']` if selectedPath is an empty string. Might cause some unexpected behaviour down the line so you should filter them out. `(selectedPath?.split('.') || []).filter(Boolean)`